src/Sylius/Behat/Context/Ui/ProductContext.php

```diff
@@ -52,4 +82,26 @@ public function iShouldNotBeAbleToAccessProduct(ProductInterface $product)
 
         expect($this->productShowPage->isOpen(['product' => $product]))->toBe(false);
     }
+
+    /**
+     * @When I delete the :product product
+     */
+    public function iDeleteProduct(ProductInterface $product)
+    {
+        $this->sharedStorage->set('product', $product);
```

Comment:
@TheMadeleine are you sure? I suppose we save a reference to an object, so if the `id` will be erased later on it will also disappear from the object we saved in shared storage since it's exactly the same object.